pants_plugins/sendwave/pants_docker/package.py

```diff
@@ -114,24 +110,26 @@ async def package_into_image(
         if component.sources:
             source_digests.append(component.sources)
         run_commands.extend(component.commands)
-    source_digest = await Get(Digest, MergeDigests(digests=source_digests))
-    application_digest = await Get(Digest, AddPrefix(source_digest, "application"))
-    snapshot = await Get(Snapshot, Digest, application_digest)
```

Comment:
You can combine these. The engine will go from AddPrefix -> Digest -> Snapshot for you.